### DISCUSSION TITLE
Infinite canvas

### DISCUSSION THREAD

**Discussion by Vadorequest** [reaviz/reaflow#51] [Q&A]
Is there a way to make the canvas "infinite"?

I understand it needs to have fixed dimensions (width/height), but I believe it's possible to dynamically increase those limits based on the content.

Is there any recommendation as to how to do that? Is it possible to know if an element is outside the canvas and automatically increase the dimensions when so? Or should we rather go for some kind of "count items" and increase the dimensions when some threshold is reached?

**Comment by amcdnl** [comment]
The `onLayoutChange` or layout prop in the ref will tell you the dimensions of the canvas - you could calculate a percentage to increase it.

On the flip side, you make a good point. I think we should make the canvas understand the size and add 20% to the base based on the size. I'll log an issue for this.

**Reply by Vadorequest** [reply]
Great! I think it should be a native feature too.

I don't have immediate need for this, so I won't implement it at this time and rather wait for an official solution.
